zulipterminal/core.py

```diff
@@ -433,13 +433,31 @@ def show_typing_notification(self) -> None:
 
         # Until conversation becomes "inactive" like when a `stop` event is sent
         while self.active_conversation_info:
-            sender_name = self.active_conversation_info["sender_name"]
-            self.view.set_footer_text(
-                [
-                    ("footer_contrast", " " + sender_name + " "),
-                    " is typing" + next(dots),
-                ]
+            active_conversation_info = ", ".join(
+                map(str, self.active_conversation_info)
             )
+            if len(self.active_conversation_info) == 1:
```

Comment:
In the following if/elif chain, there is a lot of common structure which is repeated. Try setting the footer text once, with variables that are set in the if/elif chain.

The logic will likely read clearer if you also give a name to the `len` before that logic.

Reply:
I think you can modify it even further. If you notice, only the string containing `active_conversation_info` and *is/are typing* changes across all conditions. So you can to set only those strings in the switch case, and while passing the text to `set_footer_text()`, you can provide remaining elements.

Reply:
Yep. can be better. This version should address your suggestions
```
        while self.active_conversation_info:
           active_conversation_info = ", ".join(
                map(lambda x: self.model.user_dict[x]["full_name"], self.active_conversation_info)
            )
            no_of_typing_users = len(self.active_conversation_info)
            if no_of_typing_users == 1:
                typing_phrase = " is typing"
            elif no_of_typing_users < 4:
                typing_phrase = " are typing"
            else:
                active_conversation_info = "Multiple people"
                typing_phrase = " are typing"
            typing_text = [("footer_contrast",active_conversation_info+" ")]
            typing_text.append(typing_phrase+next(dots))
            self.view.set_footer_text(typing_text)
```
the whole initial active_conversation_info part just addresses Neil's indexing through the user_id(rather than full name) suggestion.